libraries/AC_AutoTune/AC_AutoTune_FreqResp.cpp

```diff
@@ -0,0 +1,448 @@
+/*
+This library receives time history data (angular rate or angle) during a dwell test or frequency sweep test and determines the gain and phase of the response to the input. For dwell tests once the designated number of cycles are complete, the average of the gain and phase are determined over the last 5 cycles and the cycle_complete flag is set. For frequency sweep tests, phase and gain are determined for every cycle and cycle_complete flag is set to indicate when to pull the phase and gain data.  The flag is reset to enable the next cycle to be analyzed.  The init function must be used when initializing the dwell or frequency sweep test.
+*/
+
+#include <AP_HAL/AP_HAL.h>
+#include "AC_AutoTune_FreqResp.h"
+
+// Initialize the Frequency Response Object. Must be called before running dwell or frequency sweep tests
+void AC_AutoTune_FreqResp::init(InputType input_type)
+{
+    excitation = input_type;
+    max_target_cnt = 0;
+    min_target_cnt = 0;
+    max_meas_cnt = 0;
+    min_meas_cnt = 0;
+    input_start_time_ms = 0;
+    new_tgt_time_ms = 0;
+    new_meas_time_ms = 0;
+    new_target = false;
+    new_meas = false;
+    curr_test_freq = 0.0f;
+    curr_test_gain = 0.0f;
+    curr_test_phase = 0.0f;
+    max_accel = 0.0f;
+    max_meas_rate = 0.0f;
+    max_command = 0.0f;
+    meas_peak_info_buffer->clear();
```

Comment:
Theoretically at least the allocation of meas_peak_info_buffer could have failed so we might want to have a nullptr check here.  How that failure is handled more broadly is perhaps a more difficult question to answer though.

Reply:
fixed.